examples/5_secret_storage.py

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env python3
+
+###################################
+# WHAT IS IN THIS EXAMPLE?
+#
+# Keybase has added an encrypted key-value store; see 4_simple_storage.py for
+# more information.
+#
+# This example implements a simple bot to manage hackerspace tool rentals. It
+# shows one way you can obfuscate entryKeys (which are not encrypted) by
+# storing their HMACs, so that no one but your team (not even
+# Keybase) can know about the names of all the cool tools you have; you can do
+# something similar to hide namespaces. Additionally this example handles
+# concurrent writes using a cache to prevent one user from unintentionally
+# clobbering another user's rental updates.
+# ###################################
+
+import asyncio
+import hmac
+import json
+import logging
+import secrets
+import sys
+from base64 import b64decode, b64encode
+from typing import Any, Dict, List, Union
+
+from pykeybasebot import Bot, KVStoreClient
+from pykeybasebot.errors import DeleteNonExistentError, RevisionError
+from pykeybasebot.types import keybase1
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(
+        asyncio.WindowsProactorEventLoopPolicy()  # type: ignore
+    )
+
+
+class CachedBot(Bot):
+    """
+        Custom bot that has some stateful clients.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.secret_kvstore_client = SecretKeyKVStoreClient(self)
+        self.cached_secret_kvstore_client = CachedKVStoreClient(
+            self, self.secret_kvstore_client
+        )
+        super().__init__(*args, **kwargs)
+
+    @property
+    def cached_secret_kvstore(self):
+        return self.cached_secret_kvstore_client
+
+
+class CachedKVStoreClient:
+    """
+    CachedKVStoreClient uses a cache to keep track of the most recently fetched value and
+    revision for each key. To handle concurrent updates, it attempts to update with
+    the most recently fetched revision + 1; if it fails, it does a "get" and updates
+    the cache, and returns that "get" result.
+    """
+
+    def __init__(self, bot, client):
```

Comment:
it doesn't look like the CachedKVStoreClient uses `bot` for anything. which makes sense. it's a wrapper around a kvstore interface thing that gets passed in, so it doesn't really need the bot for anything.